### PR TITLE
Remove newtonsoft dependency

### PR DESCRIPTION
**Pull request by colombod**
Addressing #1010

### CODE REVIEW

File: src/Microsoft.DotNet.Interactive.Jupyter/Protocol/Message.cs [hunk 1]
```diff
@@ -74,8 +75,8 @@ public static Message TryFromJsonString(string jsonString, string messageType)
             }
 
             if (_messageTypeToClrType.TryGetValue(messageType, out var supportedType))
-            {
-                return JsonConvert.DeserializeObject(jsonString, supportedType) as Message;
+            { ;
```
Comment (brettfo) [inline src/Microsoft.DotNet.Interactive.Jupyter/Protocol/Message.cs:78]: stray semi?

File: src/Microsoft.DotNet.Interactive.PowerShell/Microsoft.DotNet.Interactive.PowerShell.csproj [hunk 2]
```diff
@@ -34,7 +34,7 @@
             CopyToOutputDirectory="PreserveNewest" />
   </ItemGroup>
 
-  <ItemGroup>
+  <!--<ItemGroup>
```
Comment (brettfo) [inline src/Microsoft.DotNet.Interactive.PowerShell/Microsoft.DotNet.Interactive.PowerShell.csproj:37]: Undo this.

File: src/Microsoft.DotNet.Interactive.Formatting.Tests/FormatterTests.cs [hunk 3]
```diff
@@ -214,27 +213,14 @@ public void Check_default_HTML_formatting_of_unicode()
                 text.Should().Be($"<div class=\"dni-plaintext\">{value.HtmlEncode()}</div>");
             }
 
-            [Fact]
-            public void When_JObject_is_formatted_it_outputs_its_string_representation()
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.Formatting.Tests/FormatterTests.cs:218]: Is there an equivalent test we should add for System.Text.Json?

File: src/Microsoft.DotNet.Interactive.Formatting.Tests/FormatterTests.cs [hunk 4]
```diff
@@ -644,21 +630,6 @@ public void Formatters_can_be_registered_on_demand_for_open_generic_interfaces(s
                 .Be(list.Count.ToString());
         }
 
-        [Fact]
-        public void When_JArray_is_formatted_it_outputs_its_string_representation()
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.Formatting.Tests/FormatterTests.cs:648]: Is there an equivalent test we should add for System.Text.Json?

File: src/Microsoft.DotNet.Interactive.Formatting.Tests/TabularDataFormatterTests.can_generate_tabular_json_from_object_array.approved.json [hunk 5]
```diff
@@ -20,27 +20,27 @@
     {
       "Name": "Q",
       "IsValid": false,
-      "Cost": 10.0
+      "Cost": 10
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.Formatting.Tests/TabularDataFormatterTests.can_generate_tabular_json_from_object_array.approved.json:23]: This is contract change that I would expect might impact the integration with the data explorer, no?

File: src/Microsoft.DotNet.Interactive.Formatting.Tests/TabularDataFormatterTests.cs [hunk 6]
```diff
@@ -27,9 +27,9 @@ public void can_generate_tabular_json_from_object_array()
             {
                 new { Name = "Q", IsValid = false, Cost = 10.0 },
                 new { Name = "U", IsValid = false, Cost = 5.0 },
-                new { Name = "E", IsValid = true, Cost = 10.0 },
-                new { Name = "S", IsValid = false, Cost = 10.0 },
-                new { Name = "T", IsValid = false, Cost = 10.0 }
+                new { Name = "E", IsValid = true, Cost = 10.2 },
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.Formatting.Tests/TabularDataFormatterTests.cs:30]: Why this change?

File: src/Microsoft.DotNet.Interactive.Formatting/DefaultJsonFormatterSet.cs [hunk 7]
```diff
@@ -1,17 +1,23 @@
...
     {
-        static internal readonly ITypeFormatter[] DefaultFormatters =
+        internal static readonly ITypeFormatter[] DefaultFormatters =
             new ITypeFormatter[]
             {
+                new JsonFormatter<string>((context, s, writer) =>
+                {
+                    var data = JsonSerializer.Serialize(s, JsonFormatter.SerializerSettings);
+                    writer.Write(data);
+                    return true;
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.Formatting/DefaultJsonFormatterSet.cs:19]: It looks like this logic is repeated insie the `JsonFormatter` class. Is any change actually needed in this file?

File: src/Microsoft.DotNet.Interactive.Formatting/JsonFormatter.cs [hunk 8]
```diff
@@ -31,7 +30,7 @@ public static ITypeFormatter GetPreferredFormatterFor(Type type)
 
         internal static ITypeFormatter[] DefaultFormatters { get; } = DefaultJsonFormatterSet.DefaultFormatters;
 
-        public static JsonSerializerSettings SerializerSettings { get; }
+        public static JsonSerializerOptions SerializerSettings { get; }
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.Formatting/JsonFormatter.cs:33]: Should probably rename the property to match the type.

File: src/Microsoft.DotNet.Interactive.Formatting.Tests/TabularDataFormatterTests.can_generate_tabular_json_from_object_array.approved.json (hunk 5 above)
Comment (colombod) [reply src/Microsoft.DotNet.Interactive.Formatting.Tests/TabularDataFormatterTests.can_generate_tabular_json_from_object_array.approved.json:23]: to ensure number serialization preserves decimals, the new serialization seems to be too smart about decimals with 0 value after the period

File: src/Microsoft.DotNet.Interactive.Formatting.Tests/TabularDataFormatterTests.cs (hunk 6 above)
Comment (colombod) [reply src/Microsoft.DotNet.Interactive.Formatting.Tests/TabularDataFormatterTests.cs:30]: to ensure number serialization preserves decimals, the new serialization seems to be too smart about decimals with 0 value after the period

File: src/Microsoft.DotNet.Interactive.Formatting/JsonFormatter.cs (hunk 8 above)
Comment (colombod) [reply src/Microsoft.DotNet.Interactive.Formatting/JsonFormatter.cs:33]: done

File: src/Microsoft.DotNet.Interactive.Jupyter/Protocol/Message.cs (hunk 1 above)
Comment (colombod) [reply src/Microsoft.DotNet.Interactive.Jupyter/Protocol/Message.cs]: eh ...

File: src/Microsoft.DotNet.Interactive.PowerShell/Microsoft.DotNet.Interactive.PowerShell.csproj (hunk 2 above)
Comment (colombod) [reply src/Microsoft.DotNet.Interactive.PowerShell/Microsoft.DotNet.Interactive.PowerShell.csproj]: ops, done

File: src/Microsoft.DotNet.Interactive.Formatting.Tests/FormatterTests.cs (hunk 3 above)
Comment (colombod) [reply src/Microsoft.DotNet.Interactive.Formatting.Tests/FormatterTests.cs:218]: we have those tests for the types in System.Text.Json

File: src/Microsoft.DotNet.Interactive.Formatting.Tests/FormatterTests.cs (hunk 4 above)
Comment (colombod) [reply src/Microsoft.DotNet.Interactive.Formatting.Tests/FormatterTests.cs:648]: we have those tests for the types in System.Text.Json

**Review by brettfo — APPROVED** [review]
